### PR TITLE
feat: Add Corpay bank info step and update file upload logic

### PR DESCRIPTION
**Pull request by pasyukevich**
### Fixed Issues

$ https://github.com/Expensify/App/issues/50898
PROPOSAL:

### Tests

Same as QA steps

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Change workspace currency to EUR, GBP, CAD or AUD
2. Enable workflows for the workspace
3. Open workflows
4. Try to connect a bank account
5. Go to the second step - a new component should be displayed
6. For  the first substep validations and field lists will be received from the BE
7. For the AUD currency we should have 3 substeps (on the second upload file step, for other currencies should be skipped)
8. Verify Upload picker validations (10 files limit, 5MB total, PDF/PNG/JPEG only)

- [x] Verify that no errors appear in the JS console
 
### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

[android-native-converted.webm](https://github.com/user-attachments/assets/284b4ec6-b8d1-49ae-a0d1-0e48307d00bb)

</details>

<details>
<summary>Android: mWeb Chrome</summary>
[android-web-converted.webm](https://github.com/user-attachments/assets/3dae1479-b1a9-4404-ad01-b37706f264a1)

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/6c7ed132-a17c-4ec9-a513-42d0c6253e20

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/e36604cc-20f9-4eae-894d-1954cec4ffe1

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/060362a0-6907-4d6d-b30a-7e9f1128fc69

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/7d63dfd6-78fd-4dfa-aa77-4b291c55d8d3

</details>

### CODE REVIEW

**Review by ishpaul777 — COMMENTED** [review]
minor comments

File: src/languages/es.ts [hunk 1]
```diff
@@ -2278,6 +2279,16 @@ const translations = {
...
     },
+    bankInfoStep: {
+        whatAreYour: '¿Cuáles son los detalles de tu cuenta bancaria comercial?',
+        letsDoubleCheck: 'Verifiquemos que todo esté bien.',
+        thisBankAccount: 'Esta cuenta bancaria se utilizará para pagos comerciales en tu espacio de trabajo.',
+        accountNumber: 'Número de cuenta',
+        bankStatement: 'Extracto bancario',
+        chooseFile: 'Elegir archivo',
+        uploadYourLatest: '¿Cuáles son los detalles de tu cuenta bancaria comercial?',
+        pleaseUpload: ({lastFourDigits}: LastFourDigitsParams) => `Por favor suba el estado de cuenta mensual más reciente de tu cuenta bancaria comercial que termina en ${lastFourDigits}.`,
```
[the remark this answers is not in the document]
Comment (pasyukevich) [reply src/languages/es.ts:2290]: Yes, they are verified

File: src/pages/ReimbursementAccount/NonUSD/BankInfo/substeps/Confirmation.tsx [hunk 2]
```diff
@@ -1,23 +1,79 @@
...
+        () => (
+            <>
+                {corpayFields.map((field) => {
+                    return (
+                        <MenuItemWithTopDescription
+                            description={field.label}
+                            title={values[field.id] ? String(values[field.id]) : ''}
+                            shouldShowRightIcon
+                            onPress={() => {
+                                if (field.id.includes(CONST.NON_USD_BANK_ACCOUNT.BANK_INFO_STEP_ACCOUNT_HOLDER_KEY_PREFIX)) {
```
[the remark this answers is not in the document]
Comment (pasyukevich) [reply src/pages/ReimbursementAccount/NonUSD/BankInfo/substeps/Confirmation.tsx]: removed

**Comment by pasyukevich** [comment]
> BUG: Upload bankstatement menu item not redirecting to upload page in confirmation page

Fixed - now should redirect as expected

**Comment by pasyukevich** [comment]
@ishpaul777 For now it is expected - it will be covered later once all steps implemented

**Comment by pasyukevich** [comment]
> Another bug it seems, when Euro is defalut currecy, the Continue bug does not redirect to next page
> 
>  Screen.Recording.2024-11-13.at.2.25.11.AM.mov

Fixed 

For some reason, this comment was not visible to me

**Comment by pasyukevich** [comment]
@ishpaul777 Now is ready for rereview 



> Ok now it work smoothly, before this was hanging too much and also crashing randomly.
> 
> One thing i have noticed is when navigating back and changing country from screen 1, it does not rest the bank account input fields and even after changing currency the input field are prefilled, it could be expected just confirming ? @pasyukevich
> 
>  Screen.Recording.2024-11-13.at.1.30.07.AM.mov

This flow will be covered later once all steps implemented

**Review by ishpaul777 — APPROVED** [review]
LGTM! 

just mention in QA steps that form validation of bank details are not proper because we used dummy fields and validations, actual data will be fetched from BE once its ready, so that QA dont create DB issue for this

**Review by madmax330 — APPROVED** [review]

**Review by madmax330 — APPROVED** [review]

File: src/libs/actions/BankAccounts.ts [hunk 3]
```diff
@@ -343,6 +343,163 @@ function validateBankAccount(bankAccountID: number, validateCode: string, policy
...
+    // TODO - Use parameters when API is ready
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const parameters = {
+        countryISO: country,
+        currency,
+        isWithdrawal: true,
+        isBusinessBankAccount: true,
+    };
+
+    // return API.read(READ_COMMANDS.GET_CORPAY_BANK_ACCOUNT_FIELDS, parameters);
```
[the remark this answers is not in the document]
Comment (pasyukevich) [reply src/libs/actions/BankAccounts.ts:356]: Yes, we will do it in the next step

**Comment by kavimuru** [comment]
@ishpaul777 @madmax330 is this PR internal PR? 
We had a similar one in the past deploy,
https://github.com/Expensify/App/pull/51667#issuecomment-2470593816

**Comment by chiragsalian** [comment]
Im checking this off the QA checklist since its holding up deploy and i think regular users won't be affected by this flow since they cannot see it atm. @madmax330, can you verify the PR on staging when you get the chance.

**Pull request merged**